src/modules/users/users.controller.ts

```diff
@@ -68,23 +71,24 @@ export class UserController {
     const limit = queryParamsDto.limit || 0;
     const skip = queryParamsDto.skip || 0;
 
-    return this.usersService.readAllUsers(
-      searchParams,
-      sortParams,
-      limit,
-      skip,
-    );
+    const impl = await this.usersFactory.getService(req.user.role);
```

Comment:
as for me here better to name var with something like `userService` instead of impl